contrib/time_series_expectations/time_series_expectations/generator/time_series_generator.py

```diff
@@ -0,0 +1,40 @@
+from abc import ABC
+from typing import TypedDict
+
+import pandas as pd
+
+
+class TrendParams(TypedDict):
+    """Parameters for a trend segment of a time series.
+
+    Each segment of a time series has a different alpha and beta value, corresponding to a different slope and intercept.
+
+    Boundaries between segments are called "cutpoints."
+    """
+
+    alpha: float
+    beta: float
+    cutpoint: int
+
+
+class TimeSeriesGenerator(ABC):
+    """Base class for time series generators."""
+
+    def generate_df(
+        self,
+        *args,
+        **kwargs,
+    ) -> pd:
+        """Generate a time series as a pandas DataFrame.
+
+        Args:
+            *args will differ depending on the specific generator class.
+
+        Keyword Args:
+            *kwargs will differ depending on the specific generator class.
+
+        Returns:
+            pd.DataFrame: A two-column pandas DataFrame with a datetime index and a column for the time series values
+
+        """
+        raise NotImplementedError()
```

Comment:
Non-blocking comment.  When we extend ABC, we can also `import  from abc abstractmethod` and then decorate `generate_df()` with `@abstractmethod` (then the `generate_df()` does not need to raise `NotImplementedError()` but can simple contain `pass`).  Thank you.

Reply:
Got it. I'll make that change before merging.